src/QuadForm/Quad/ZLattices.jl

```diff
@@ -934,3 +934,26 @@ function invariant_lattice(L::ZLat, G::MatElem;
                            ambient_representation::Bool = true)
   return kernel_lattice(L, G - 1, ambient_representation = ambient_representation)
 end
+
+################################################################################
+#
+#  Membership check
+#
+################################################################################
+
+@doc Markdown.doc"""
+    Base.in(v::Vector, L::ZLat) -> Bool
+
+
+  This function checks if the vector 'v' lies in the lattice 'L' or not.
+"""
+function Base.in(v::Vector, L::ZLat)
+  @assert size(v)[1]==rank(L) "The vector should have the same length as the rank of the lattice."
+  B = basis_matrix(L)
+  V = matrix(QQ, size(v)[1], 1, v)
+  if !isone(denominator(V)) || !can_solve(B, V)
```

Comment:
`[1//2]` is in the lattice with basis matrix `[1//2;]` but has denominator not one.

Reply:
That is true. I didn't think about the non-integer cases. What would you recommend? I can add an option to make it possible for users to just get the answer for the integral case. Or should I just remove the denominator check altogether?

Reply:
You can call `fl, w = can_solve_with_solution(B, V)`. Then `fl` must be true and `w` must be integral. If one of those is not true, then `v` is not in the lattice.

Reply:
I wonder if we want to cache the hnf of L? And then use `can_solve_left_reduced_triu`?

Reply:
Who knows that this function is doing. I would not try to do it in this PR here.